src/main/java/com/github/dockerjava/jaxrs/BuildImageCmdExec.java

```diff
@@ -75,10 +76,8 @@ private static AuthConfigurations firstNonNull(final AuthConfigurations fromComm
             webTarget = webTarget.queryParam("t", command.getTag());
         }
 
-        if (command.getCacheFrom() != null) {
-            for (String c: command.getCacheFrom()) {
-               webTarget = webTarget.queryParam("cachefrom", c);
-            }
+        if (command.getCacheFrom() != null && !command.getCacheFrom().isEmpty()) {
+            webTarget = webTarget.queryParam("cachefrom", "[\"" + Joiner.on("\",\"").join(command.getCacheFrom()) + "\"]");
```

Comment:
Is it the same as https://github.com/docker-java/docker-java/blob/master/src/main/java/com/github/dockerjava/core/exec/ListServicesCmdExec.java#L30

Reply:
Please reuse existing helper methods, + parameter should be escaped

Reply:
@KostyaSha good point! It's not a Map, but I created a similar utility class